multimap/src/main/java/org/infinispan/multimap/api/MultimapCache.java

```diff
@@ -0,0 +1,226 @@
+ * <h2>Duplicates</h2> Today's implementation does not support duplicate values on keys. {@link Object#equals(Object)}
+ * method is used to check if a value is already present in the key. This means that the following code
+ * <pre>
+ *    multimapCache.put("k", "v1").join();
+ *    multimapCache.put("k", "v2").join();
+ *    multimapCache.put("k", "v2").join();
+ *    multimapCache.put("k", "v2").join();
+ *
+ *    multimapCache.get("k").thenAccept(values -> System.out.println(values.size()));
+ * </pre>
+ * prints the value 2. "k" -> ["v1", "v"]
+ * <p>
+ * Future implementations might evolve to support duplicated values.
+ * <p>
+ * <h2>Transactions</h2> MultimapCache supports implicit transactions without blocking. The following methods block when
+ * they are called in a explicit transaction context. This limitation could be improved in the followin versions if
+ * technically possible. <ul> <li>{@link MultimapCache#size()}</li> <li>{@link MultimapCache#containsEntry(Object,
+ * Object)}</li> <li>{@link MultimapCache#remove(Predicate)}</li> <li></li> </ul> More about transactions in : <a
+ * href="http://infinispan.org/docs/dev/user_guide/user_guide.html#transactions">the Infinispan Documentation</a>.
+ *
+ * @author Katia Aresti, karesti@redhat.com
+ * @see <a href="http://infinispan.org/documentation/">Infinispan documentation</a>
+ * @since 9.2
+ */
+@Experimental
+public interface MultimapCache<K, V> {
+
+   /**
+    * Asynchronous method. Puts a key-value pair in this multimap cache.
+    * <ul>
+    * <li>If this multimap cache supports duplicates, the value will be always added.</li>
+    * <li>If this multimap cache does <i>not support</i> duplicates and the value exists on the key,
+    * nothing will be done.</li>
+    * </ul>
+    *
+    * @param key,   the key to be put
+    * @param value, the value to added
+    * @return {@link CompletableFuture} containing a {@link Void}
+    * @since 9.2
+    */
+   CompletableFuture<Void> put(K key, V value);
+
+   /***
+    * Asynchronous method. Returns a <i>view collection</i> of the values associated with key in this multimap cache,
+    * if any. Any changes to the retrieved collection won't change the values in this multimap cache.
+    * <b>When this method returns an empty collection, it means the key was not found.</b>
+    *
+    * @param key, to be retrieved
+    * @return a {@link CompletableFuture} containing {@link Collection <V>} which is a view of the underlying values.
+    * @since 9.2
+    */
+   CompletableFuture<Collection<V>> get(K key);
+
+   /**
+    * Retrieves a CacheEntry corresponding to a specific key in this multimap cache.
+    *
+    * @param key the key whose associated cache entry is to be returned
+    * @return the cache entry to which the specified key is mapped, or
+    *         {@code null} if this multimap contains no mapping for the key
+    *
+    * @since 9.2
+    */
+   CompletableFuture<Optional<CacheEntry<K, Collection<V>>>> getEntry(K key);
```

Comment:
I copied the java doc on entry and changed after to optional, didn't realize, thanks for pointing